decidim-core/app/helpers/decidim/newsletters_helper.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Decidim
+  # Helper that provides methods to render order selector and links
+  module NewslettersHelper
+    def parse_interpolations(content, user = nil, id = nil)
+      if id.present? && user.present?
+        host = user.organization.host.to_s
+        campaign = "newsletter_#{id}"
+
+        links = content.scan(/href\s*=\s*"([^"]*)"/)
+
+        links.each do |link|
+          link_replaced = link.first + utm_codes(host, campaign)
```

Comment:
Appending tracking codes should be optional. Until we have an admin interface to configure these things, this could be set in `Decidim.track_newsletter_links = true` in `config/initializers/decidim.rb`.